### PR TITLE
Batch shift

### PR DESCRIPTION
**Pull request by Zhaoyi-Yan**

### CODE REVIEW

File: models/shift_net/InnerShiftTripleFunction.py [hunk 1]
```diff
@@ -30,34 +29,53 @@ def forward(ctx, input, shift_sz, stride, triple_w, flag, show_flow):
...
+        # These three lines of code work similarly with `non_mask_indexes_tmp[indexes]`
+        # However, pytorch advance indexing dose not support multi-dimensional indexing.
+        non_mask_indexes_tmp = (flag==0).nonzero()[:,1].view(ctx.bz, -1)
+        idx_tmp = indexes + torch.arange(indexes.size(0)).view(-1,1).type_as(indexes) * non_mask_indexes_tmp.size(1)
+        non_mask_indexes = non_mask_indexes_tmp.view(-1)[idx_tmp]
 
-            # GET FINAL SHIFT FEATURE
-            shift_masked_all[idx] = Nonparm._paste(latter_windows, ctx.ind_lst[idx], i_2, i_3, i_1, i_4)
+        # Update: batch op if someone think up an approach
+        for i in range(ctx.bz):
```
Comment (Zhaoyi-Yan) [inline models/shift_net/InnerShiftTripleFunction.py]: @tchaton It is possible to remove `for loop` here ? It should be the final snippet that isn't implemented in batch version.

**Comment by Zhaoyi-Yan** [comment]
It gives the same results of the origin code(both in forward and backward) and is faster when batchsize is large(0.049s/image --> 0.025s/image when batchsize=64)

File: models/shift_net/InnerShiftTripleFunction.py (hunk 1 above)
Comment (tchaton) [reply models/shift_net/InnerShiftTripleFunction.py]: @Zhaoyi-Yan , I think it is possible to vectorize everything there.

Comment (Zhaoyi-Yan) [reply models/shift_net/InnerShiftTripleFunction.py]: @tchaton I have ask in the pytorchDiscuss and get a nice solution. I have updated it.

**Pull request merged**